core/src/main/java/org/apache/brooklyn/location/byon/ByonLocationResolver.java

```diff
@@ -199,8 +199,8 @@ protected ConfigBag extractConfig(Map<?,?> locationFlags, String spec, LocationR
         }
         
         Class<? extends MachineLocation> locationClassHere = locationClass;
-        if (osfamily != null) {
-            locationClassHere = getLocationClass(osfamily);
+        if (osFamily != null) {
+            locationClassHere = OS_TO_MACHINE_LOCATION_TYPE.get(osFamily);
```

Comment:
This reverts @sjcorbett 's change in commit c3e027b80826427d04e3b726bf9f71f24fc46103. Can you change it back to use `getLocationClass(osFamily)` please.